references/optical_flow/utils.py

```diff
@@ -0,0 +1,390 @@
+import datetime
+import os
+import time
+from collections import defaultdict
+from collections import deque
+
+import torch
+import torch.distributed as dist
+import torch.nn.functional as F
+
+
+class SmoothedValue:
```

Comment:
This, and the `MetricLogger` below have been copy/pasted from the classification references. I only made some very minor changes like setting some defaults, probably not worth reviewing.